ui/src/pages/AccountCreation/CreateAccount.js

```diff
@@ -0,0 +1,246 @@
+import { useEffect, useState } from 'react';
+import "../../App.css";
+import { appWindow, LogicalSize } from '@tauri-apps/api/window';
+import { Outlet } from 'react-router-dom';
+import { useNavigate, useLocation } from "react-router-dom";
+
+const MIN_PASSWORD_LENGTH = 8;
+
+const DEFAULT_WINDOW_SIZE = new LogicalSize(460, 500);
+const CONFIRM_PHRASE_WINDOW_SIZE = new LogicalSize(460, 600);
+
+const CreateAccount = ({
+  sendSelection,
+  recoveryPhrase,
+  sendPassword,
+  restartServer
+}) => {
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [passwordsMatch, setPasswordsMatch] = useState(false);
+  const [isValidPassword, setIsValidPassword] = useState(false);
+  const [recoveryPhraseConfirmed, setRecoveryPhraseConfirmed] = useState(false);
+  const [isValidSelectedPhrase, setIsValidSelectedPhrase] = useState(false);
+  const [shuffledRecoveryPhrase, setShuffledRecoveryPhrase] = useState(null);
+  const [selectedRecoveryPhrase, setSelectedRecoveryPhrase] = useState([]);
+  const [actualPhrase, setActualPhrase] = useState(null);
+  const [showError, setShowError] = useState(false);
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const checkPasswords = () => {
+
+    if (!passwordsMatch || !isValidPassword) {
+      setShowError(true);
+    } else {
+      goForward();
+    }
+
+  }
+
+  useEffect(() => {
+
+    const checkPasswordValidity = () => {
+      if (isValid(password)) {
+        setIsValidPassword(true);
+      } else {
+        setIsValidPassword(false);
+      }
+    }
+  
+    const checkPasswordMatch = () => {
+      if (password === confirmPassword) {
+        setPasswordsMatch(true);
+      } else {
+        setPasswordsMatch(false);
+      }
+    }
+
+    checkPasswordMatch();
+    checkPasswordValidity();
+
+  }, [password, confirmPassword, isValidPassword, passwordsMatch]);
```

Comment:
I don't think you need `isValidPassword` or `passwordsMatch` as dependencies. Normally if you are setting something in a `useEffect`, you don't list it as a dependency. This can cause an endless loop or rerenders.

Reply:
Yea, I forgot to remove these, I was getting eslint warnings, however I will just add a line to ignore them for this case.